### PR TITLE
Nc feat/update toolbar menu dropdown with new design

### PR DESCRIPTION
**Pull request by dstala**
## Change Summary

- ref: #7638

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

## Summary by CodeRabbit

- **New Features**
    - Introduced a new size option 'xsmall' for Switch components.
    - Added a searchable field list component with grouping, sorting, and searching capabilities.
    - Implemented drag and drop functionality in the Fields Menu with enhanced UI.
    - Enhanced toolbar dropdown search fields with improved styling and functionality.
- **Style**
    - Fixed syntax errors and reorganized styling rules across the application.
    - Updated scrollbar, dropdown menus, select options, icons, and button styling for a cohesive look.
    - Adjusted styles for various toolbar components and added new style definitions for thinner scrollbars.
- **Bug Fixes**
    - Fixed search value reset issues in toolbar components.
    - Refined search functionality with updated column handling and selection logic.
- **Refactor**
    - Refactored styling and structure for better readability and maintainability in multiple components.
    - Updated Vue component props and template logic for enhanced user interaction.
- **Documentation**
    - No visible changes to end-users.

### CODE REVIEW

File: packages/nc-gui/components/smartsheet/toolbar/FieldsMenu.vue [hunk 1]
```diff
@@ -300,6 +300,19 @@ const showSystemField = computed({
   },
 })
 
+const isDragging = ref<boolean>(false)
+
```
[the remark this answers is not in the document]
Comment (rameshmane7218) [reply packages/nc-gui/components/smartsheet/toolbar/FieldsMenu.vue:304]: It is not necessary to add comment on every single variable as variable name clearly tells that it is used to store value of dragging item

File: packages/nc-gui/components/smartsheet/toolbar/CreateGroupBy.vue [hunk 2]
```diff
@@ -71,6 +71,7 @@ watch(
   () => {
     if (!isParentOpen.value) return
 
+    search.value = ''
```
[the remark this answers is not in the document]
Comment (rameshmane7218) [reply packages/nc-gui/components/smartsheet/toolbar/CreateGroupBy.vue:74]: Tried but it's not working

[the remark this answers is not in the document]
Comment (rameshmane7218) [reply packages/nc-gui/components/smartsheet/toolbar/CreateGroupBy.vue:74]: Yes, I know that but we are using a-input from antdv UI library so may be it's taking time to load or focus input, so it's important to add timout there

File: packages/nc-gui/components/smartsheet/toolbar/FieldListWithSearch.vue [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+  // As we need to focus search box when the parent is opened
+  isParentOpen: boolean
+  toolbarMenu: 'groupBy' | 'sort' | 'globalSearch'
+  searchInputPlaceholder?: string
+  selectedOptionId?: string
+  options: ColumnType[]
+  showSelectedOption?: boolean
+}>()
+
+const emits = defineEmits(['selected'])
```
Comment (dammy001) [inline packages/nc-gui/components/smartsheet/toolbar/FieldListWithSearch.vue:14]:
```suggestion
const emits = defineEmits<{ selected: [ColumnType]}>()
```

**Review by rameshmane7218 — APPROVED** [review]

**Pull request merged**